dali/operators/decoder/inflate/inflate_params.h

```diff
@@ -0,0 +1,397 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_DECODER_INFLATE_INFLATE_PARAMS_H_
+#define DALI_OPERATORS_DECODER_INFLATE_INFLATE_PARAMS_H_
+
+#include <string>
+#include <vector>
+
+#include "dali/core/backend_tags.h"
+#include "dali/core/common.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/pipeline/data/types.h"
+#include "dali/pipeline/operator/arg_helper.h"
+#include "dali/pipeline/operator/operator.h"
+
+namespace dali {
+
+namespace inflate {
+
+constexpr static const char *dTypeArgName = "dtype";
+constexpr static const char *algArgName = "algorithm";
+constexpr static const char *shapeArgName = "shape";
+constexpr static const char *offsetArgName = "chunks_offsets";
+constexpr static const char *sizeArgName = "chunks_sizes";
+constexpr static const char *layoutArgName = "layout";
+constexpr static const char *sequenceLayoutArgName = "sequence_extent";
+
+enum class InflateAlg {
+  LZ4
+};
+
+inline std::string to_string(const InflateAlg &alg) {
+  switch (alg) {
+    case InflateAlg::LZ4:
+      return "LZ4";
+    default:
+      return "<unknown>";
+  }
+}
+
+inline InflateAlg parse_inflate_alg(std::string str) {
+  std::transform(str.begin(), str.end(), str.begin(), [](auto c) { return std::tolower(c); });
+  if (str == "lz4") {
+    return InflateAlg::LZ4;
+  }
+  DALI_FAIL(make_string("Unknown inflate algorithm was specified for `", algArgName,
+                        "` argument: `", str, "`."));
```

Comment:
I don't think there are multiple arguments where we can specify an algorithm. Also, don't use inverted quotes for a string.
```suggestion
  DALI_FAIL(make_string("Unknown inflate algorithm: \"", str, "\"."));
```

Reply:
I follwed requirements specified by @klecki that forsee supporting more compression algorithms. It seems reasonable.

Reply:
More algorithms - yes. What I meant is multiple _arguments_. Besides, the message doesn't look good:

> Unknown inflate algorithm was specified for 'algorithm' argument: 'wtf'

I think the shorter one conveys just as much information:

> Unknown inflate algorithm 'wtf'

Reply:
Oh, sure, now I get your point. I've changed the error message.